code/obj/machinery/sleeper.dm

```diff
@@ -545,41 +545,43 @@ TYPEINFO(/obj/machinery/sleeper)
 				move_inside()
 			else
 				return
-		else if (can_operate(user))
+		else if (can_operate(user,target))
 			var/previous_user_intent = user.a_intent
 			user.set_a_intent(INTENT_GRAB)
 			user.drop_item()
 			target.Attackhand(user)
 			user.set_a_intent(previous_user_intent)
 			SPAWN(user.combat_click_delay + 2)
-				if (can_operate(user))
+				if (can_operate(user,target))
 					if (istype(user.equipped(), /obj/item/grab))
 						src.Attackby(user.equipped(), user)
 
-	proc/can_operate(var/mob/M)
-		if (!(BOUNDS_DIST(src, M) == 0))
+	proc/can_operate(var/mob/M, var/mob/living/target)
```

Comment:
This proc call exists behind a `SPAWN` you should verify M and target still exist and arent disposed before continuing.

However because of how many mobs don't cleanly garbage collect a simple `if (M && target)` will not suffice. You will instead want to make use of the `QDELETED()` macro